pkg/reconciler/buildrun/resources/volumes.go

```diff
@@ -0,0 +1,85 @@
+// Copyright The Shipwright Contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package resources
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func namespacedName(name, namespace string) types.NamespacedName {
+	return types.NamespacedName{
+		Namespace: namespace,
+		Name:      name,
+	}
+}
+
+// CheckTaskRunVolumesExist tries to find some of the volumes referenced by the BuildRun with all the
+// overrides. If some secret or configmap does not exist in the namespace, function returns error
+// describing the missing resource
+func CheckTaskRunVolumesExist(ctx context.Context, client client.Client, taskRun *v1beta1.TaskRun) error {
+	for _, volume := range taskRun.Spec.TaskSpec.Volumes {
+		var (
+			err          error
+			resourceType string
+			name         string
+		)
+
+		switch {
+		case volume.Secret != nil:
+			resourceType = "Secret"
+			secret := corev1.Secret{}
+			name = volume.Secret.SecretName
+			err = client.Get(ctx, namespacedName(name, taskRun.Namespace), &secret)
+			if !apierrors.IsNotFound(err) {
+				err = nil
+			}
+		case volume.ConfigMap != nil:
+			resourceType = "ConfigMap"
+			configMap := corev1.ConfigMap{}
+			name = volume.ConfigMap.Name
+			err = client.Get(ctx, namespacedName(name, taskRun.Namespace), &configMap)
```

Comment:
Don't we need `get` then in 200-role instead of `list`?

Reply:
I got an error about missing `list` right, and when I added it, it got fixed.
I suppose there is a different error which is not `resource not found`:
```
			if !apierrors.IsNotFound(err) {
				err = nil
			}
```
We are not interested in configMap contents anyways, only whether it exists or not.